test/quantization/test_numeric_suite_fx.py

```diff
@@ -429,3 +443,117 @@ def test_compare_model_outputs_lstm_dynamic_fx(self):
+        # verify that matching graphs with matching node types but
+        # different counts of matchable nodes fails
+        m1 = nn.Sequential(nn.Conv2d(1, 1, 1)).eval()
+        m2 = nn.Sequential(nn.Conv2d(1, 1, 1), nn.Conv2d(1, 1, 1)).eval()
+        mp1 = prepare_fx(m1, {'': torch.quantization.default_qconfig})
+        mp2 = prepare_fx(m2, {'': torch.quantization.default_qconfig})
+        with self.assertRaises(GraphMatchingException) as ex:
+            results = get_matching_node_pairs(mp1, mp2)
+
+    def test_matching_failure_node_type(self):
+        # verify that matching graphs with non-matching node types fails
+        m1 = nn.Sequential(nn.Conv2d(1, 1, 1)).eval()
+        m2 = nn.Sequential(nn.Linear(1, 1)).eval()
+        mp1 = prepare_fx(m1, {'': torch.quantization.default_qconfig})
+        mp2 = prepare_fx(m2, {'': torch.quantization.default_qconfig})
+        with self.assertRaises(GraphMatchingException) as ex:
+            results = get_matching_node_pairs(mp1, mp2)
+
+    def test_conv_multilayer_mod_fp32_prepared_vs_int8(self):
+        m = nn.Sequential(
+            nn.Sequential(
+                nn.Conv2d(1, 1, 1),
+            ),
+            nn.Conv2d(1, 1, 1),
+        ).eval()
+        mp = prepare_fx(m, {'': torch.quantization.default_qconfig})
+        # TODO(future PR): prevent the need for copying here, we can copy the
+        # modules but should reuse the underlying tensors
+        mp_copy = copy.deepcopy(mp)
+        mq = convert_fx(mp_copy)
+        # assume success if no exceptions
+        results = get_matching_node_pairs(mp, mq)
+
+    def test_tensor_ops_fp32_prepared_vs_int8(self):
+        class M(nn.Module):
+            def __init__(self):
+                super().__init__()
+
+            def forward(self, x, y):
+                z = x + y
+                return z
+
+        m = M().eval()
+        mp = prepare_fx(m, {'': torch.quantization.default_qconfig})
+        # TODO(future PR): prevent the need for copying here, we can copy the
+        # modules but should reuse the underlying tensors
+        mp_copy = copy.deepcopy(mp)
+        mq = convert_fx(mp_copy)
+        # assume success if no exceptions
+        results = get_matching_node_pairs(mp, mq)
+
+    @skip_if_no_torchvision
+    def test_mobilenet_v2_fp32_prepared_vs_int8(self):
+        # verify that mobilenetv2 graph is able to be matched
+        import torchvision
+        m = torchvision.models.__dict__['mobilenet_v2'](pretrained=False).eval().float()
+        mp = prepare_fx(m, {'': torch.quantization.default_qconfig})
+        # TODO(future PR): prevent the need for copying here, we can copy the
+        # modules but should reuse the underlying tensors
+        mp_copy = copy.deepcopy(mp)
+        mq = convert_fx(mp_copy)
+        # assume success if no exceptions
+        results = get_matching_node_pairs(mp, mq)
+
+    @skip_if_no_torchvision
+    def test_mobilenet_v2_fp32_qat_prepared_vs_int8(self):
```

Comment:
For later PRs, can we also check matching for fp32 vs prepared and prepared QAT?

Reply:
yes, definitely